server/src/weaverbird/backends/pandas_executor/steps/utils/formula.py

```diff
@@ -15,14 +15,26 @@
 }
 
 
+class NullSeries(Series):
+    """
+    When the expr is "null", this is a float series filled with NaN.
+    This subclass allow us to keep this information for necessary casts, e.g in ifthenelse step.
+    """
+
+    def __init__(self, index: DataFrame.index):
+        Series.__init__(self, index=index, dtype="object")
```

Comment:
```suggestion
        super().__init__(self, index=index, dtype="object")
```

Why not an `object` Series filled with `None` ? You're already using the `object` dtype. I don't see the advantage of a float series here

Reply:
Ah, my bad, I forgot the `None`. 
What do you mean by `a float series`?

Reply:
it's what you said in the docstring:
> When the expr is "null", this is a float series filled with NaN.

Reply:
Oh, good catch! Docstring is lying 😄 

In old pandas version, building a Series with None create a float series by default, with a warning to tell us dtype="object" is now the way to go. 

This is exactly what I did, and it even fixed a cast issue. 

Fixed the docstring in [2496b3e](https://github.com/ToucanToco/weaverbird/pull/2422/commits/2496b3e1ad052aadd2b27068f786c21a90b355a4).

---

The suggestion about `super()` gives this error: 'NullSeries' object has no attribute '_name'
I'm not proficient enough in python to know what is going on.

Reply:
Ah, `Series` probably has a metaclass. You can leave it as it is